File: backend/src/main/java/edu/cit/spot/entity/Attendance.java
package edu.cit.spot.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
@Data
@Table(name = "attendances")
public class Attendance {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "section_id", nullable = false)
    private Section section;
    
    @NotNull
    private LocalDate date;
    
    @NotNull
    @Column(name = "start_time")
    private LocalTime startTime;
    
    @Column(name = "end_time")
    private LocalTime endTime;
}

File: frontend_mobile/app/src/main/java/com/example/spot/model/Attendance.kt
package com.example.spot.model

/**
 * Attendance data model matching the backend AttendanceDto
 */
data class Attendance(
    val id: Long,
    val student: Student,
    val section: Section,
    val date: String, // YYYY-MM-DD format
    val startTime: String?, // Using String instead of LocalTime to match backend JSON format
    val endTime: String?    // Using String instead of LocalTime to match backend JSON format
)
